cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureDatabaseResourceServiceTest.java

```diff
@@ -86,6 +89,9 @@ class AzureDatabaseResourceServiceTest {
     @Mock
     private AzureClient client;
 
+    @Mock
```

Comment:
I tried to mock `client.getAzure().genericResources().getById(cloudResource.getReference()).inner().identity().principalId()` - but couldn't successfully do it. How to mock `genericResources` ?

@lajosrodek Please suggest some way to get though this. I'll add the required testcase accordingly.

Reply:
Hello Juhi, what was the problem when mocking the genericResources? 
I quickly checked and it worked, despite Azure being a final class - mock-maker-inline helps to overcome this problem.
My quick test code was just as below:

```
        Azure azure = mock(Azure.class);
        GenericResources genericResources = mock(GenericResources.class);
        when(azure.genericResources()).thenReturn(genericResources);
        GenericResource genericResource = mock(GenericResource.class);
        when(genericResource.resourceGroupName()).thenReturn("myRgName");
        when(genericResources.getById("myResourceName")).thenReturn(genericResource);
        when(client.getAzure()).thenReturn(azure);
```

Reply:
@gergopapi2 Thank you for inputs. I have updated the test :)